internal/adapters/api/handler/role.go

```diff
@@ -0,0 +1,64 @@
+package handler
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/iBoBoTi/project_boiler_plate/internal/adapters/api/response"
+	"github.com/iBoBoTi/project_boiler_plate/internal/core/domain"
+	"github.com/iBoBoTi/project_boiler_plate/internal/core/ports"
+	"net/http"
+)
+
+type roleHandler struct {
+	roleService       ports.RoleService
+	permissionService ports.PermissionService
+	logger            ports.Logger
+}
+
+func NewRoleHandler(roleService ports.RoleService, permissionService ports.PermissionService, logger ports.Logger) ports.RoleHandler {
+	return &roleHandler{
+		roleService:       roleService,
+		permissionService: permissionService,
```

Comment:
is there any reason why you need your permission service here in your handler.If you even need them it should be injected inside your roleService

Reply:
Removed it, initially I wanted to create an additional endpoint for the Role Handler for adding permissions to a role before you told me to create handlers for role_permissions